src/components/Editor/CustomExtensions/CodeBlock/CodeBlockComponent.jsx

```diff
@@ -1,23 +1,78 @@
-import React from "react";
+import React, { useState } from "react";
 
 import { NodeViewWrapper, NodeViewContent } from "@tiptap/react";
 import { copyToClipboard } from "neetocommons/utils";
-import { Copy } from "neetoicons";
-import { Button } from "neetoui";
-
-const CodeBlockComponent = ({ node }) => (
-  <NodeViewWrapper data-cy="neeto-editor-code-block">
-    <pre>
-      <Button
-        className="copy-button"
-        icon={Copy}
-        size="small"
-        style="text"
-        onClick={() => copyToClipboard(node?.content?.content[0]?.text)}
-      />
-      <NodeViewContent as="code" />
-    </pre>
-  </NodeViewWrapper>
-);
+import { Copy, Down } from "neetoicons";
+import { Button, Dropdown, Input } from "neetoui";
+import { equals } from "ramda";
+import { useTranslation } from "react-i18next";
+
+import { SORTED_LANGUAGE_LIST } from "./constants";
+
+const { Menu, MenuItem } = Dropdown;
+
+const CodeBlockComponent = ({ node, updateAttributes }) => {
+  const [keyword, setKeyword] = useState("");
+
+  const { t } = useTranslation();
+
+  const filteredAndSortedLanguageList = SORTED_LANGUAGE_LIST.filter(language =>
+    language.includes(keyword)
+  );
+
+  const handleCopyClick = () =>
+    copyToClipboard(node?.content?.content[0]?.text);
+
+  const handleLanguageSelect = language => {
+    updateAttributes({ language: equals(language, "auto") ? null : language });
+    setKeyword("");
+  };
+
+  return (
+    <NodeViewWrapper data-cy="neeto-editor-code-block">
+      <pre>
+        <div className="neeto-editor-codeblock-options">
+          <Dropdown
+            buttonSize="small"
+            buttonStyle="secondary"
+            closeOnOutsideClick={false}
+            icon={Down}
+            label={node.attrs?.language || t("common.auto")}
```

Comment:
```suggestion
          <Dropdown
            strategy="fixed"
            buttonSize="small"
            buttonStyle="secondary"
            closeOnOutsideClick={false}
            icon={Down}
            label={node.attrs?.language || t("common.auto")}
```

This will make the dropdown appear above the editor:
![image](https://github.com/bigbinary/neeto-editor/assets/35297280/0505ad47-ffce-4e50-af7c-3d1aaaf48795)